source/extensions/filters/common/rbac/engine_impl.cc

```diff
@@ -11,7 +11,8 @@ namespace Common {
 namespace RBAC {
 
 RoleBasedAccessControlEngineImpl::RoleBasedAccessControlEngineImpl(
-    const envoy::config::rbac::v3::RBAC& rules, const EnforcementMode mode)
+    const envoy::config::rbac::v3::RBAC& rules, const EnforcementMode mode,
+    const absl::optional<ProtobufMessage::ValidationVisitor*>& validation_visitor)
```

Comment:
Why is this optional? Where are you missing a validation visitor?

Reply:
Mostly for convenience.  This would get passed on to PolicyMatcher  and  eventually to `AndMatcher` and `OrMatcher` whose usage  might not have any handle to the visitor.  Currently the test `test/extensions/filters/common/rbac/matchers_test.cc` is an example of how matchers are used without a validator.

Reply:
Let's make this required for simplicity, thanks.